src/internal/packager/deploy.go

```diff
@@ -350,22 +350,33 @@ func (p *Packager) pushImagesToRegistry(componentImages []string, noImgChecksum
 }
 
 // Push all of the components git repos to the configured git server
-func (p *Packager) pushReposToRepository(reposPath string, repos []string) {
+func (p *Packager) pushReposToRepository(reposPath string, repos []string) error {
 	if len(repos) == 0 {
-		return
+		return nil
 	}
+	totalFailed := 0
 
 	// Try repo push up to 3 times
-	for retry := 0; retry < 3; retry++ {
-		// Push all the repos from the extracted archive
-		if err := git.New(p.cfg.State.GitServer).PushAllDirectories(reposPath); err != nil {
-			message.Errorf(err, "Unable to push repos to the Git Server, retrying in 5 seconds...")
-			time.Sleep(5 * time.Second)
-			continue
-		} else {
-			break
-		}
+	for _, repoPath := range repos {
+		gitClient := git.New(p.cfg.InitOpts.GitServer)
+
+		utils.Retry(func() error {
+			if totalFailed < 3 {
```

Comment:
I know it is "one more thing" to add to this refactor but thoughts on making the retry amount a const in config?  This would make it easier to introduce a command flag to tune the number of retries if that is needed on some systems.

Reply:
It may even be nice to pass a function into a retry helper that is more centralized too so that retry logic can be the same everywhere.